util/make_timeline.py

```diff
@@ -134,9 +188,62 @@ def parse_file(args):
 
     return entries, last_ability_time
 
+def find_fights_in_file(file):
+    e_starts = []
+    encounter_start_staging = 0 # Staged to avoid spurious starts such as limit break usage
+    e_ends = []
+    encounter_in_progress = False
+    current_instance = None
+    encounter_sets = []
+    for line in file:
+        # Ignore log lines that aren't game status info
+        if line[37:41] != '0839' and line[0:2] not in ['00', '01', '21', '22', '33']:
+            continue
+        line_fields = line.split('|')
+
+        # Update current instance for returning alongside timestamps
+        # Don't update if we're not entering a combat instance
+        if is_instance_begun(line_fields):
+            current_instance = line_fields[4].split(' has begun')[0]
+            continue
+        if is_instance_ended(line_fields):
+            current_instance = None
+            encounter_in_progress = False
+            continue
+
+        # We don't want to look for encounters outside a combat instance
+        if current_instance is None:
+            continue
+
+        # Build start/end time groupings
+        if is_line_start(line_fields, encounter_in_progress):
```

Comment:
I find bools passed to helper functions a little bit confusing.  What do you think about rewriting this block as:
```
if is_zone_seal(line_fields):
    encounter_start_staging = ...
    encounter_in_progress = True
    continue
elif not encounter_in_progress and is_line_attack(line_fields):
    encounter_start_staging = ...
    encounter_in_progress = True
    continue
```

and getting rid of is_line_start? In other words, hoist the logic out of is_line_start.  I think as a helper it maybe encapsulates too much logic and so is confusing, and it'd be better to be explicit at the callsite.  What do you think?

Reply:
That's perfectly fair. Un-nesting that `if` is nice too.